src/commands.rs

```diff
@@ -1744,20 +1770,19 @@ fn json_obj_keys<M>(redis_key: &mut M::ReadHolder, path: &str) -> RedisResult
 where
     M: Manager,
 {
-    let root = redis_key.get_value()?;
-    let res: RedisValue = match root {
-        Some(root) => {
-            let values = find_all_values(path, root, |v| v.get_type() == SelectValueType::Object)?;
-            let mut res: Vec<RedisValue> = vec![];
-            for v in values {
-                res.push(match v {
-                    Some(v) => v.keys().unwrap().collect::<Vec<&str>>().into(),
-                    _ => RedisValue::Null,
-                });
-            }
-            res.into()
+    let root = redis_key
+        .get_value()?
+        .ok_or_else(RedisError::nonexistent_key)?;
+    let res: RedisValue = {
```

Comment:
Before we handled the case where root is None, don't we need to handle it anymore?

Reply:
Now it is handled in `json_obj_keys` and `json_obj_keys_legacy`